private/bufpkg/bufmodule/bufmoduleref/util.go

```diff
@@ -20,25 +20,25 @@ import (
 )
 
 // parseModuleReferenceComponents parses and returns the remote, owner, repository,
-// and ref (branch or commit) from the given path.
+// and ref (branch, commit, draft, or tag) from the given path.
 func parseModuleReferenceComponents(path string) (remote string, owner string, repository string, ref string, err error) {
```

Comment:
Style nit, feel free to ignore: this makes sense to me, logically, but would it be possible/reasonable to refactor this not to use named returns? We're not using the default zero values and explicitly setting zero values in error return cases anyway, I think we might as well instantiate these explicitly in the function...

Reply:
as per internal style guide, I think we prefer named returns when the function returns multiple values of the same type to make it more readable, and the default zero values of `ref` is actually used in the case of `len(split) == 1`, so I think it's reasonable to keep this as is?

Reply:
That seems very reasonable to me!